facades/TeraEd/src/main/java/org/terasology/editor/subsystem/LwjglPortletDisplayDevice.java

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.editor.subsystem;
+
+import org.lwjgl.glfw.GLFW;
+import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.awt.AWTGLCanvas;
+import org.terasology.engine.subsystem.DisplayDevice;
+import org.terasology.engine.subsystem.DisplayDeviceInfo;
+import org.terasology.engine.subsystem.Resolution;
+import org.terasology.engine.subsystem.lwjgl.LwjglDisplayDevice;
+import org.terasology.engine.subsystem.lwjgl.LwjglGraphicsManager;
+import org.terasology.engine.subsystem.lwjgl.LwjglGraphicsUtil;
+import org.terasology.engine.subsystem.lwjgl.LwjglResolution;
+import org.terasology.rendering.nui.layers.mainMenu.videoSettings.DisplayModeSetting;
+import org.terasology.utilities.subscribables.AbstractSubscribable;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LwjglPortletDisplayDevice extends AbstractSubscribable implements DisplayDevice {
+
+    private final AWTGLCanvas canvas;
+    private final LwjglGraphicsManager graphics;
+    private DisplayDeviceInfo displayDeviceInfo;
+
+    public LwjglPortletDisplayDevice(AWTGLCanvas canvas, LwjglGraphicsManager graphics) {
+        this.canvas = canvas;
+        this.graphics = graphics;
+        canvas.addComponentListener(new ComponentAdapter() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                updateViewport();
+            }
+        });
+        graphics.asynchToDisplayThread(() -> {
+            displayDeviceInfo = LwjglGraphicsUtil.getDisplayDeviceInfo();
```

Comment:
This data can be changed by resize?

Reply:
The reason for this is, that it calls GL11 functions which requires active OpenGL context to process, so have to be done in display thread. There are currently vendor strings, so it should be ok to call once (unless someone can switch graphic cards in the middle of playing?)

Reply:
Hmm it maybe interesting- change graphics during teraed session. But not in resize event.

Reply:
Well, it could be red every render tick, so that it will be available for metrics 1-tick delayed in the worst case.

Reply:
Ideally render ticks should works every 1/60 seconds.
then 1-tick delay is not afraid :)